### DISCUSSION TITLE
How to get byte_index from the byte array to read real values ?

### DISCUSSION THREAD

**Discussion by niravpatidar37** [gijzelaerr/python-snap7#285] [Q&A] [answered]
plc = snap7.client.Client()
plc.connect('192.168.1.101', 0,1)
all_data = plc.read_area(S7AreaDB,1,0,14)
print(all_data[:])

real=snap7.util.get_real(all_data,byte_index)
print(real)

need a byte index to get the value of real type variable in byte array so how to find byte index of that ?

**Comment by lautarodapin** [comment, marked as answer]
You need to know in advance the structure of the DB that you will read, otherwise you are blind
Example:
In tia portal you have a db like this

| Datattpe | byte.bit |
-------------------------------
| Bool        | 0.0         |
| Bool        | 0.1         |

...
Real 2.0

For reading the real 2.0 you will do
```py
get_real(all_data, 2)
```
